plugins/autolink/plugin.js

```diff
@@ -127,6 +121,10 @@
 				return query.match( editor.config.autolink_urlRegex ) ||
 					query.match( editor.config.autolink_emailRegex );
 			}
+		},
+
+		isSupportedEnvironment: function() {
```

Comment:
`isSupportedEnvironment` is a kind of plugin metadata, it should be placed right after the plugin `requires` property.

Reply:
Moved `isSupportedEnvironment` to the top of the file:  https://github.com/ckeditor/ckeditor4/pull/2518/commits/b6fd74b101e83c6b360cdb5c5ad7be256632522c